From ddc05360bd41c65efd8f86e5ef3a129f6c64c7ef Mon Sep 17 00:00:00 2001
From: Caesar Wang <wxt@rock-chips.com>
Date: Sat, 24 Feb 2018 11:49:41 +0800
Subject: [PATCH] media: soc_camera/rockchip: Fix the build error

As enable the releated config, hit the below error:

drivers/media/i2c/soc_camera/rockchip/adv_camera_module.c:937:34: error:
    assignment to expression with array type
        timings->exposure_valid_frame =
...

and

drivers/media/i2c/soc_camera/rockchip/tc_camera_module.c:963:34: error:
    assignment to expression with array type
        timings->exposure_valid_frame =
....

Change-Id: I4381cc8aa00bb6a968c01d7537a8942b5715fa5c
Signed-off-by: Caesar Wang <wxt@rock-chips.com>
---
 drivers/media/i2c/soc_camera/rockchip/adv_camera_module.c | 2 +-
 drivers/media/i2c/soc_camera/rockchip/tc_camera_module.c  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/i2c/soc_camera/rockchip/adv_camera_module.c b/drivers/media/i2c/soc_camera/rockchip/adv_camera_module.c
index f6f08c9d7978..cbc9a7d4d3bd 100644
--- a/drivers/media/i2c/soc_camera/rockchip/adv_camera_module.c
+++ b/drivers/media/i2c/soc_camera/rockchip/adv_camera_module.c
@@ -934,7 +934,7 @@ long adv_camera_module_ioctl(struct v4l2_subdev *sd,
 			adv_timings.fine_integration_time_min;
 
 		if (cam_mod->custom.g_exposure_valid_frame)
-			timings->exposure_valid_frame =
+			timings->exposure_valid_frame[0] =
 				cam_mod->custom.g_exposure_valid_frame(cam_mod);
 		if (cam_mod->exp_config.exp_time)
 			timings->exp_time = cam_mod->exp_config.exp_time;
diff --git a/drivers/media/i2c/soc_camera/rockchip/tc_camera_module.c b/drivers/media/i2c/soc_camera/rockchip/tc_camera_module.c
index 50b02c10a4c1..f862b6799591 100644
--- a/drivers/media/i2c/soc_camera/rockchip/tc_camera_module.c
+++ b/drivers/media/i2c/soc_camera/rockchip/tc_camera_module.c
@@ -960,7 +960,7 @@ long tc_camera_module_ioctl(struct v4l2_subdev *sd,
 			tc_timings.fine_integration_time_min;
 
 		if (cam_mod->custom.g_exposure_valid_frame)
-			timings->exposure_valid_frame =
+			timings->exposure_valid_frame[0] =
 				cam_mod->custom.g_exposure_valid_frame(cam_mod);
 		if (cam_mod->exp_config.exp_time)
 			timings->exp_time = cam_mod->exp_config.exp_time;
-- 
2.35.3

